web/api/servers.go

```diff
@@ -978,6 +989,215 @@ func editServerDataAdmin(c *gin.Context) {
 	proxyServerRequest(c)
 }
 
+// @Summary Gets servers backups
+// @Description Gets all backups made on this server
+// @Success 200 {object} models.Backup
+// @Param id path string true "Server ID"
+// @Router /api/servers/{id}/backup [get]
+// @Security OAuth2Application[server.backup.view]
+func getBackups(c *gin.Context) {
+	server := getServerFromGin(c)
+	db := middleware.GetDatabase(c)
+	bs := &services.Backup{DB: db}
+
+	records, err := bs.GetAllBakcupsForServer(server.Identifier)
+
+	if response.HandleError(c, err, http.StatusInternalServerError) {
+	} else {
+		c.JSON(http.StatusOK, records)
+	}
+}
+
+// @Summary Create backup
+// @Description Creates a full backup of the server
+// @Success 204 {object} nil
+// @Param id path string true "Server ID"
+// @Query name query string true "name of the backup"
+// @Router /api/servers/{id}/backup/create [post]
+// @Security OAuth2Application[server.backup.create]
+func createBackup(c *gin.Context) {
+	server := getServerFromGin(c)
+	db := middleware.GetDatabase(c)
+	ns := &services.Node{DB: db}
+	bs := &services.Backup{DB: db}
+	name := c.Query("name")
+	node := &server.Node
+
+	resolvedPath := "/daemon/server/" + strings.TrimPrefix(c.Request.URL.Path, "/api/servers/")
+	if c.Request.URL.RawQuery != "" {
+		resolvedPath += "?" + c.Request.URL.RawQuery
+	}
+
+	callResponse, err := ns.CallNode(node, c.Request.Method, resolvedPath, c.Request.Body, c.Request.Header)
+	if callResponse.StatusCode == http.StatusBadRequest { //If its a local node, the err will not be set, have to check the status code
+		newHeaders := cleanHttpReturnErrors(callResponse.Header)
+
+		c.DataFromReader(callResponse.StatusCode, callResponse.ContentLength, callResponse.Header.Get("Content-Type"), callResponse.Body, newHeaders)
+		c.Abort()
+		return
+	}
+	if response.HandleError(c, err, http.StatusInternalServerError) {
+		return
+	}
+
+	responseData := &pufferpanel.ServerBackupResponse{}
+	err = json.NewDecoder(callResponse.Body).Decode(responseData)
+	if response.HandleError(c, err, http.StatusBadRequest) {
+		return
+	}
+
+	backup := &models.Backup{Name: name, FileName: responseData.BackupFileName, FileSize: responseData.FileSize, ServerID: server.Identifier}
+	err = bs.Create(backup)
+	if response.HandleError(c, err, http.StatusBadRequest) {
+		return
+	}
+
+	c.Status(http.StatusOK)
```

Comment:
If there is no body, use StatusNoContent instead

Reply:
fixed in [3afa64c](https://github.com/pufferpanel/pufferpanel/pull/1379/commits/3afa64cabbf2a66b294418fec744d28d5024bb52)